src/Projection/InMemoryProjectionManager.php

```diff
@@ -112,48 +112,63 @@ public function stopProjection(string $name): void
         throw new Exception\RuntimeException('Stopping a projection is not supported in ' . get_class($this));
     }
 
-    public function fetchProjectionNames(?string $filter, bool $regex, int $limit, int $offset): array
+    public function fetchProjectionNames(?string $filter, int $limit = 20, int $offset = 0): array
     {
-        if (null === $filter && $regex) {
-            throw new Exception\InvalidArgumentException('No regex pattern given');
+        if (0 >= $limit) {
+            throw new Exception\OutOfRangeException(
+                'Invalid limit "'.$limit.'" given. Must be greater than 0.'
+            );
         }
 
-        if ($regex && false === @preg_match("/$filter/", '')) {
-            throw new Exception\InvalidArgumentException('Invalid regex pattern given');
+        if (0 > $offset) {
+            throw new Exception\OutOfRangeException(
+                'Invalid offset "'.$offset.'" given. Must be greater or equal than 0.'
+            );
         }
 
-        $result = [];
+        if (null === $filter) {
+            $result = array_keys($this->projections);
+            $result = array_slice($result, $offset, $limit);
+            sort($result, \SORT_NATURAL);
```

Comment:
Also we need to sort first, before we slice the result, otherwise you get different results.